iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/GeneratorModification.java

```diff
@@ -40,29 +42,30 @@ public void apply(Network network, boolean throwException,
         if (modifs.getMaxP() != null) {
             g.setMaxP(modifs.getMaxP());
         }
-
-        if (modifs.getVoltageRegulatorOn() != null) {
-            g.setVoltageRegulatorOn(modifs.getVoltageRegulatorOn());
+        if (modifs.getTargetV() != null) {
+            g.setTargetV(modifs.getTargetV());
+        }
+        if (modifs.getTargetQ() != null) {
+            g.setTargetQ(modifs.getTargetQ());
         }
-
         boolean skipOtherConnectionChange = false;
         if (modifs.getConnected() != null) {
             changeConnectionState(g, modifs.getConnected());
             skipOtherConnectionChange = true;
         }
-
+        if (modifs.getVoltageRegulatorOn() != null) {
+            if (Double.isNaN(g.getTargetV())) {
```

Comment:
I think we should not change the targetV if the voltage regulation is set to OFF:
```suggestion
            if (modifs.getVoltageRegulatorOn().booleanValue() && Double.isNaN(g.getTargetV())) {
```

Reply:
Yes, right of course.